apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useWatchWalletSync.ts

```diff
@@ -0,0 +1,191 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useDispatch, useSelector, useStore } from "react-redux";
+import { CloudSyncSDK, UpdateEvent } from "@ledgerhq/live-wallet/cloudsync/index";
+import walletsync, {
+  liveSlug,
+  DistantState,
+  walletSyncWatchLoop,
+  LocalState,
+  Schema,
+} from "@ledgerhq/live-wallet/walletsync/index";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { useTrustchainSdk } from "./useTrustchainSdk";
+import { walletSelector } from "~/renderer/reducers/wallet";
+import { MemberCredentials, Trustchain, TrustchainSDK } from "@ledgerhq/trustchain/types";
+import {
+  memberCredentialsSelector,
+  trustchainSelector,
+  setTrustchain,
+  resetTrustchainStore,
+} from "@ledgerhq/trustchain/store";
+import { TrustchainEjected } from "@ledgerhq/trustchain/errors";
+import { State } from "~/renderer/reducers";
+import { cache as bridgeCache } from "~/renderer/bridge/cache";
+import { setAccountNames, walletSyncUpdate } from "@ledgerhq/live-wallet/store";
+import { replaceAccounts } from "~/renderer/actions/accounts";
+import noop from "lodash/noop";
+import { log } from "@ledgerhq/logs";
+
+function localStateSelector(state: State): LocalState {
+  return {
+    accounts: {
+      list: state.accounts,
+    },
+    accountNames: state.wallet.accountNames,
+  };
+}
+
+function latestDistantStateSelector(state: State): DistantState | null {
+  return walletSelector(state).wsState.data;
+}
+
+export type WalletSyncUserState = {
+  visualPending: boolean;
+  walletSyncError: Error | null;
+  onUserRefresh: () => void;
+};
+
+export function useCloudSyncSDK(): CloudSyncSDK<Schema> {
+  const trustchainSdk = useTrustchainSdk();
+
+  const store = useStore();
+  const dispatch = useDispatch();
+
+  const getCurrentVersion = useCallback(() => {
+    const state = walletSelector(store.getState());
+    return state.wsState.version;
+  }, [store]);
+
+  const saveNewUpdate = useCallback(
+    async (event: UpdateEvent<DistantState>) => {
+      log("walletsync", "saveNewUpdate", { event });
+      switch (event.type) {
+        case "new-data": {
```

Comment:
can we type `event.type` ?

Reply:
mmh it is typed (as a union of static strings), if you do `case "foo": `  it will produce an error.

![Capture d’écran 2024-07-15 à 16 22 55](https://github.com/user-attachments/assets/48296d0c-fb80-467e-95f0-d84729da6b19)